server/backend/message.go

```diff
@@ -19,6 +19,8 @@ type message struct {
 	labelIDs   []string
 	sysLabel   *string
 	attIDs     []string
+	references []string
```

Comment:
I don't think this needs to be stored in the server. If we have the `inReplyTo` (which you could probably call `parentID`, like `externalID`), then the references header can be created at message build time in `getHeader()` with some simple logic (checking if it's non-empty).

Reply:
I found it more explicit to have it here to highlight what the live API is doing (or what this value reflects).
I feel like putting it in the `getHeader()`  will somehow "hide" how they are handled in the backend.

Reply:
But this is what he API does -- it doesn't store the references as a separate DB entry, it generates them on demand in the header during send based on existing parameters (parentID). I think what i suggest actually matches API's behaviour more closely.

Reply:
Done